tests/generation/test_utils.py

```diff
@@ -1236,6 +1236,50 @@ def test_prompt_lookup_decoding_matches_greedy_search(self):
             for output in (output_greedy, output_prompt_lookup):
                 self._check_outputs(output, input_ids, model.config, use_cache=True)
 
+    def test_dola_decoding_sample(self):
+        # TODO (joao): investigate skips, try to reduce incompatibilities
+        for model_class in self.all_generative_model_classes:
+            if any(model_name in model_class.__name__.lower() for model_name in ["reformer"]):
+                self.skipTest("Skip Reformer as the lm_head input size is 2 * hidden size, adopted from Rev Nets.")
+
+            if any(model_name in model_class.__name__.lower() for model_name in ["marian", "mbart", "pegasus"]):
+                self.skipTest("DoLa is not supported for models that don't return layerwise hidden states")
+
+            # enable cache if the model is not openai-gpt, xlnet, cpm, or xlm
+            config, input_ids, attention_mask, _ = self._get_input_ids_and_config(batch_size=1)
+
+            if not hasattr(config, "use_cache"):
+                config.use_cache = False
+            else:
+                config.use_cache = True
```

Comment:
Based on https://github.com/huggingface/transformers/pull/29619/files#r1538243054

```suggestion
            # Some models don't support the cache and returning past_key_values
            if not hasattr(config, "use_cache"):
                config.use_cache = False
            else:
                config.use_cache = True
```

Reply:
Added the comment!